dclab/rtdc_dataset/writer.py

```diff
@@ -209,15 +208,32 @@ def store_feature(self, feat, data, shape=None):
                         + f"you could put the shape `{data[0].shape}` "
                         + 'in the `info["feature shapes"]` key of '
                         + "your plugin feature.")
-                    shape = data.shape[1:]
-            if shape == data.shape:
-                data = data.reshape(1, *shape)
-            elif shape == data.shape[1:]:
-                pass
+            # Condition for ragged/contour data (array of arrays)
+            if data.ndim == 1 and isinstance(data[0], np.ndarray):
+                self.write_ragged(group=events, name=feat, data=data)
+
+            # Condition for array of scalar features
+            elif data.ndim == 1 and not isinstance(data[0], np.ndarray):
+                self.write_ndarray(group=events, name=feat, data=data)
```

Comment:
This is happening because scalar PF's are not saved by this line. They are saved by the condition above (line number 174 to 177). I got to know this via debugging. Thank you for telling me how to make use of debugging!

Reply:
The issue is that the line isn't being tested though, so when you have fixed this, please resolve this thread

Reply:
It is fixed now.